cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreEndpointContextInfo.java

```diff
@@ -0,0 +1,207 @@
+     * @param request the request to set the endpoint context for
+     */
+    public static void receivingRequest(OSCoreCtx oscoreCtx, Request request) {
+        setInfoIncoming(oscoreCtx, request);
+    }
+
+    /**
+     * Sets information in a destination endpoint context for outgoing responses.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param response the response to set the endpoint context for
+     */
+    public static void sendingResponse(OSCoreCtx oscoreCtx, Response response) {
+        setInfoOutgoing(oscoreCtx, response);
+    }
+
+    /**
+     * Sets information in a destination endpoint context for incoming responses.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param response the response to set the endpoint context for
+     */
+    public static void receivingResponse(OSCoreCtx oscoreCtx, Response response) {
+        setInfoIncoming(oscoreCtx, response);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * destination endpoint context for outgoing messages.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param message the message to set information in the destination endpoint context for
+     */
+    private static void setInfoOutgoing(OSCoreCtx oscoreCtx, Message message) {
+
+        //Do nothing if this is not an instance of MapBasedEndpointContext (must cast to it)
+        if (message.getDestinationContext() instanceof MapBasedEndpointContext == false) {
+            return;
+        }
+
+        //Create new MapBasedEndpointContext for destination endpoint context with string values for OSCORE added
+        MapBasedEndpointContext endpointContext = (MapBasedEndpointContext) message.getDestinationContext();
+
+        MapBasedEndpointContext newEndpointContext = setInfo(oscoreCtx, endpointContext);
+
+        message.setDestinationContext(newEndpointContext);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * source endpoint context for incoming messages.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param message the message to set information in the source endpoint context for
+     */
+    private static void setInfoIncoming(OSCoreCtx oscoreCtx, Message message) {
+
+        //Do nothing if this is not an instance of MapBasedEndpointContext (must cast to it)
+        if (message.getSourceContext() instanceof MapBasedEndpointContext == false) {
+            return;
+        }
+
+        //Create new MapBasedEndpointContext for source endpoint context with string values for OSCORE added
+        MapBasedEndpointContext endpointContext = (MapBasedEndpointContext) message.getSourceContext();
+
+        MapBasedEndpointContext newEndpointContext = setInfo(oscoreCtx, endpointContext);
+
+        message.setSourceContext(newEndpointContext);
```

Comment:
Consider to use:

```
EndpointContext newEndpointContext = setInfo(oscoreCtx, message.getSourceContext());
message.setSourceContext(newEndpointContext);
```

That should do it. No cast nor "instanceof" check.

Reply:
Yes, thanks for the suggestion. That did it with no cast or instanceof. I just had to add a check for null in the beginning of setInfo, otherwise I get a null pointer exception for some JUnit tests at https://github.com/eclipse/californium/blob/11ccb133c0356d97816e2ab1b4ba7d6dc7ece24c/element-connector/src/main/java/org/eclipse/californium/elements/MapBasedEndpointContext.java#L211

Done in commit d0c4594169ec6a56545192005ef395b2f69cf501